--- a/original_file.c
+++ b/modified_file.c
@@ -17,6 +17,7 @@
     int i, j, jajp1, nza, k, nzrow;
     double xi;
     nrows = lastrow - firstrow + 1;
+#pragma omp parallel for
     for (j = 1; j <= n; j++) {
 	rowstr[j] = 0;
 	mark[j] = FALSE;
@@ -30,7 +31,8 @@
     for (j = 2; j <= nrows+1; j++) {
 	rowstr[j] = rowstr[j] + rowstr[j-1];
     }
-      for(j = 0;j <= nrows-1;j++) {
+#pragma omp parallel for private(k)
+    for(j = 0;j <= nrows-1;j++) {
          for(k = rowstr[j];k <= rowstr[j+1]-1;k++)
 	       a[k] = 0.0;
       }
@@ -47,6 +49,7 @@
     }
     rowstr[1] = 1;
     nza = 0;
+#pragma omp parallel for
     for (i = 1; i <= n; i++) {
 	x[i] = 0.0;
 	mark[i] = FALSE;