--- a/tzetar.c
+++ b/tzetar.c
@@ -2,6 +2,7 @@
   int i, j, k;
   double t1, t2, t3, ac, xvel, yvel, zvel, r1, r2, r3,
     r4, r5, btuz, acinv, ac2u, uzik1;
+#pragma omp parallel for private(j, k, t1, t2, t3, ac, xvel, yvel, zvel, r1, r2, r3, r4, r5, btuz, acinv, ac2u, uzik1)
   for (i = 1; i <= grid_points[0]-2; i++) {
     for (j = 1; j <= grid_points[1]-2; j++) {
       for (k = 1; k <= grid_points[2]-2; k++) {