--- a/ep.c
+++ b/ep.c
@@ -68,6 +68,7 @@
 
     k_offset = -1;
 
+#pragma omp parallel default(shared) reduction(+:sx, +:sy)
 {
     double t1, t2, t3, t4, x1, x2;
     int kk, i, ik, l;
@@ -78,6 +79,7 @@
     for (i = 0; i < NQ; i++) qq[i] = 0.0;
 
     for (k = 1; k <= np; k++) {
+#pragma omp for
 	kk = k_offset + k;
 	t1 = S;
 	t2 = an;
@@ -97,6 +99,7 @@
 	if (TIMERS_ENABLED == TRUE) timer_stop(3);
     }
     {
+#pragma omp critical
       for (i = 0; i <= NQ - 1; i++) q[i] += qq[i];
     }
 #if defined(_OPENMP)