--- a/code.c
+++ b/code.c
@@ -27,6 +27,7 @@
   if (ifin1 == ii2) ifin1 = ifin -1;
   if (jfin1 == ji2) jfin1 = jfin -1;
   for (i = 0; i <= ISIZ2+1; i++) {
+#pragma omp parallel for collapse(2) private(i, k)
     for (k = 0; k <= ISIZ3+1; k++) {
       phi1[i][k] = 0.0;
       phi2[i][k] = 0.0;
@@ -34,6 +35,7 @@
   }
   for (i = ibeg; i <= ifin; i++) {
     iglob = i;
+#pragma omp parallel for collapse(2) private(i, j, iglob, jglob)
     for (j = jbeg; j <= jfin; j++) {
       jglob = j;
       k = ki1;
@@ -54,6 +56,7 @@
 #endif
   frc1 = 0.0;
   for (i = ibeg; i <= ifin1; i++) {
+#pragma omp parallel for collapse(2) private(i, j) reduction(+:frc1)
     for (j = jbeg; j <= jfin1; j++) {
       frc1 = frc1 + (  phi1[i][j]
 		       + phi1[i+1][j]
@@ -67,6 +70,7 @@
   }
   for (i = 0; i <= ISIZ2+1; i++) {
     for (k = 0; k <= ISIZ3+1; k++) {
+#pragma omp parallel for collapse(2) private(i, k)
       phi1[i][k] = 0.0;
       phi2[i][k] = 0.0;
     }
@@ -75,6 +79,7 @@
   jglob = jbeg;
   if (jglob == ji1) {
     for (i = ibeg; i <= ifin; i++) {
+#pragma omp parallel for collapse(2) private(i, k, iglob)
       iglob = i;
       for (k = ki1; k <= ki2; k++) {
 	phi1[i][k] = C2*(  u[i][jbeg][k][4]
@@ -88,6 +93,7 @@
   jglob = jfin;
   if (jglob == ji2) {
     for (i = ibeg; i <= ifin; i++) {
+#pragma omp parallel for collapse(2) private(i, k, iglob)
       iglob = i;
       for (k = ki1; k <= ki2; k++) {
 	phi2[i][k] = C2*(  u[i][jfin][k][4]
@@ -100,6 +106,7 @@
 #endif
   frc2 = 0.0;
   for (i = ibeg; i <= ifin1; i++) {
+#pragma omp parallel for collapse(2) private(i, k) reduction(+:frc2)
     for (k = ki1; k <= ki2-1; k++) {
       frc2 = frc2 + (  phi1[i][k]
 		       + phi1[i+1][k]
@@ -113,6 +120,7 @@
   }
   for (i = 0; i <= ISIZ2+1; i++) {
     for (k = 0; k <= ISIZ3+1; k++) {
+#pragma omp parallel for collapse(2) private(i, k)
       phi1[i][k] = 0.0;
       phi2[i][k] = 0.0;
     }
@@ -121,6 +129,7 @@
   iglob = ibeg;
   if (iglob == ii1) {
     for (j = jbeg; j <= jfin; j++) {
+#pragma omp parallel for collapse(2) private(j, k, jglob)
       jglob = j;
       for (k = ki1; k <= ki2; k++) {
 	phi1[j][k] = C2*(  u[ibeg][j][k][4]
@@ -134,6 +143,7 @@
   iglob = ifin;
   if (iglob == ii2) {
     for (j = jbeg; j <= jfin; j++) {
+#pragma omp parallel for collapse(2) private(j, k, jglob)
       jglob = j;
       for (k = ki1; k <= ki2; k++) {
 	phi2[j][k] = C2*(  u[ifin][j][k][4]
@@ -146,6 +156,7 @@
 #endif
   frc3 = 0.0;
   for (j = jbeg; j <= jfin1; j++) {
+#pragma omp parallel for collapse(2) private(j, k) reduction(+:frc3)
     for (k = ki1; k <= ki2-1; k++) {
       frc3 = frc3 + (  phi1[j][k]
 		       + phi1[j+1][k]