--- a/original_code.c
+++ b/parallelized_code.c
@@ -6,6 +6,7 @@
   double  pxi, peta, pzeta;
   double  ue_1jk[5],ue_nx0jk[5],ue_i1k[5],
     ue_iny0k[5],ue_ij1[5],ue_ijnz[5];
+#pragma omp parallel for private(i, j, k, m, iglob, jglob, xi, eta, zeta, pxi, peta, pzeta, ue_1jk, ue_nx0jk, ue_i1k, ue_iny0k, ue_ij1, ue_ijnz)
   for (j = 0; j < ny; j++) {
     jglob = j;
     for (k = 1; k < nz - 1; k++) {