--- a/ssor.c
+++ b/ssor.c
@@ -6,6 +6,7 @@
   double  delunm[5], tv[ISIZ1][ISIZ2][5];
   tmp = 1.0 / ( omega * ( 2.0 - omega ) ) ;
 {
+#pragma omp parallel for collapse(2)
   for (i = 0; i < ISIZ1; i++) {
     for (j = 0; j < ISIZ2; j++) {
       for (k = 0; k < 5; k++) {
@@ -31,6 +32,7 @@
     }
 {
     for (i = ist; i <= iend; i++) {
+#pragma omp parallel for
       for (j = jst; j <= jend; j++) {
 	for (k = 1; k <= nz - 2; k++) {
 	  for (m = 0; m < 5; m++) {
@@ -59,6 +61,7 @@
     }
     for (i = ist; i <= iend; i++) {
       for (j = jst; j <= jend; j++) {
+#pragma omp parallel for
 	for (k = 1; k <= nz-2; k++) {
 	  for (m = 0; m < 5; m++) {
 	    u[i][j][k][m] = u[i][j][k][m]