--- a/original_file.c
+++ b/modified_file.c
@@ -1,11 +1,13 @@
 static void y_solve_cell(void) {
   int i, j, k, jsize;
+
   jsize = grid_points[1]-1;
+  #pragma omp parallel for private(i, k)
   for (i = 1; i < grid_points[0]-1; i++) {
     for (k = 1; k < grid_points[2]-1; k++) {
       binvcrhs( lhs[i][0][k][BB],
 		lhs[i][0][k][CC],
-		rhs[i][0][k] );
+		rhs[i][0][k] );
     }
   }
   for (j = 1; j < jsize; j++) {
@@ -24,6 +26,7 @@
     }
   }
   for (i = 1; i < grid_points[0]-1; i++) {
+    #pragma omp parallel for private(i, k)
     for (k = 1; k < grid_points[2]-1; k++) {
       matvec_sub(lhs[i][jsize][k][AA],
 		 rhs[i][jsize-1][k], rhs[i][jsize][k]);