--- a/original_file.c
+++ b/modified_file.c
@@ -1,6 +1,7 @@
 static void z_solve_cell(void) {
   int i,j,k,ksize;
   ksize = grid_points[2]-1;
+  #pragma omp parallel for private(j)
   for (i = 1; i < grid_points[0]-1; i++) {
     for (j = 1; j < grid_points[1]-1; j++) {
       binvcrhs( lhs[i][j][0][BB],
@@ -10,6 +11,7 @@
   }
   for (k = 1; k < ksize; k++) {
       for (i = 1; i < grid_points[0]-1; i++) {
+	#pragma omp parallel for private(j)
 	  for (j = 1; j < grid_points[1]-1; j++) {
 	matvec_sub(lhs[i][j][k][AA],
 		   rhs[i][j][k-1], rhs[i][j][k]);
@@ -24,6 +26,7 @@
     }
   }
   for (i = 1; i < grid_points[0]-1; i++) {
+    #pragma omp parallel for private(j)
     for (j = 1; j < grid_points[1]-1; j++) {
       matvec_sub(lhs[i][j][ksize][AA],
 		 rhs[i][j][ksize-1], rhs[i][j][ksize]);