--- a/original_file.c
+++ b/modified_file.c
@@ -1,9 +1,10 @@
 static void compute_indexmap(int indexmap[NZ][NY][NX], int d[3]) {
     int i, j, k, ii, ii2, jj, ij2, kk;
     double ap;
+#pragma omp parallel for collapse(3) private(i, j, k, ii, ii2, jj, ij2, kk)
     for (i = 0; i < dims[2][0]; i++) {
 	ii =  (i+1+xstart[2]-2+NX/2)%NX - NX/2;
-	ii2 = ii*ii;
+ ii2 = ii*ii;
 	for (j = 0; j < dims[2][1]; j++) {
             jj = (j+1+ystart[2]-2+NY/2)%NY - NY/2;
             ij2 = jj*jj+ii2;